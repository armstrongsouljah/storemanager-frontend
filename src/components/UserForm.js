import React from 'react';
import TextInput from './TextInput';

const UserForm = props => (
    
            <div className="">
                <h4 className="center">Add New User</h4>
                {
                    props.onLoading ?
                    <div class="preloader-wrapper active">
                    <div class="spinner-layer spinner-red-only">
                      <div class="circle-clipper left">
                        <div class="circle"></div>
                      </div><div class="gap-patch">
                        <div class="circle"></div>
                      </div><div class="circle-clipper right">
                        <div class="circle"></div>
                      </div>
                    </div>
                  </div>
                    :
                    ''
                }
                <form className="form section" onSubmit={props.onSignup}>
                    <TextInput id="username" type="text" labelText="Username" onChange={props.onChange} />
                    <TextInput id="password" type="password" labelText="Password" onChange={props.onChange} />
                    <div className="input-field">
                    <label>User role</label>
                    <br/><br/>
                        <select className="browser-default" id="role" onChange={props.onSelectChange}>
                            <option value="" defaultValue>=========</option>
                            <option value="admin" defaultValue>Admin</option>
                            <option value="attendant">Attendant</option>
                        </select>
                </div>
                    <button className="btn cyan darken-4" type="submit" name="action">
Save
                        <i className="material-icons right">send</i>
                    </button>
                    <br/> <br/>
                    <span className="red-text">{props.errors? props.errors: ''} </span>
                    <span className="teal-text">{props.successMessage ? props.successMessage: ''} </span>
                </form>
            </div>
);

export default UserForm;
