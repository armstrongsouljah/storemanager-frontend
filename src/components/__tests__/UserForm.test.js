import React from 'react'; 
import { shallow } from 'enzyme';

import UserForm from '../UserForm';

describe('USerForm', () => {
    it('renders wtihout crashing', () => {
        const wrapper = shallow(<UserForm/>);
        wrapper.setProps({
            errors:'',
            successMessage:''
        })
        expect(wrapper).toMatchSnapshot()
    })
})
