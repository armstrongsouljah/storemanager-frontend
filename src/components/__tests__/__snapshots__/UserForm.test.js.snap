// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USerForm renders wtihout crashing 1`] = `ShallowWrapper {}`;
